### DISCUSSION TITLE
HTTP/3 Support

### DISCUSSION THREAD

**Discussion by baseplate-admin** [django/daphne#399] [Ideas]
Hi there,

Is there any plan to add HTTP/3 Support ? [Hypercorn supports HTTP/3](https://github.com/pgjones/hypercorn) and theres a python [module](https://github.com/aiortc/aioquic/) that adds support. I can work on this. But before that i want to know if a Pull Request be worth it?

**Comment by carltongibson** [comment]
Hi @baseplate-admin — _What's involved?_ (What's the twisted story here?) TBH I haven't thought about it at all, so some initial discussion is needed. In general _sounds great yes_ but what does that really mean?

(Going to convert to a discussion to begin.)

**Reply by baseplate-admin** [reply]
Hi there,

Thanks for moving it into a discussion.

>  What's involved?

Adding a dependency([aioquic](https://github.com/aiortc/aioquic/))  . Preferably with ``daphne[http3]`` and somehow ( i didn't take a look) patch it.

> What does that really mean?

Daphne will add support for [HTTP/3 Protocol](https://datatracker.ietf.org/doc/html/draft-ietf-quic-http-18). ( meaning the QUIC protocol support )

> What's the twisted story here?

We'll have to ask Twisted at upstream to support HTTP/3 :) or patch it here to add HTTP/3.

> what does that really mean?

A lot of hours looking at daphne and twisted source code and trying to figure out what and where to patch :P

### Why should we support it ?

Because it will be the future standard. Also if you have a lil bit of time you can [read the differences between HTTP/2 and HTTP/3](https://blog.cloudflare.com/http-3-vs-http-2/) (by cloudflare)


### Another Note :
Also take a look [here](https://github.com/encode/uvicorn/discussions/1105) ( by tomchristie )
The discussion seems to state that **python community doesn't have much desire for Python HTTP/2 or HTTP/3 servers but a fast HTTP/1.1 one.**  At this rate, (_**Daphne is the slowest asgi server**_) Daphne wont see much adoption outside Django Community ( even django community is shifting towards other web server : ```eg: uvicorn( for speed ) ``` ). So in order to make Daphne mainstream again, I think we should :

1. Support HTTP/3 ( to compete with Hypercorn )
2. Make Daphne Faster ( to compete with Uvicorn ) ( maybe add a Gunicorn worker class ? Like Uvicorn does )

**A web server with Daphne's security, Hypercorn's feature, Uvicorn's speed will be a killer combo**. _But maybe its just my pipe dream and will never be implemented fully_.

Thanks for reading and Happy New Year.
